ql/src/test/org/apache/hadoop/hive/ql/hooks/TestHiveHookEventProtoPartialBuilder.java

```diff
@@ -49,6 +58,15 @@ public void testEquality() {
     assertArrayEquals(event1.toByteArray(), event2.toByteArray());
   }
 
+  @Test
+  public void testOtherInfoQueryPlan() {
+    String stageIdRearrange = HiveConf.ConfVars.HIVE_STAGE_ID_REARRANGE.defaultStrVal;
+    HiveHookEvents.HiveHookEventProto event = buildIn2Steps(mock(JSONObject.class), getWork(), stageIdRearrange);
+    // Without HIVE-29012, OTHERINFO will have CONF initially i.e size is 1.
+    // With HIVE-29012, OTHERINFO should have QUERY as well i.e size is 2.
+    assert(event.getOtherInfo(1).hasKey());
```

Comment:
I believe that using a java assertion here was not intentional; please replace this with a junit assertion. 

In addition, consider using `assertEquals` with the full `List` instead of assertTrue/False. The failure message will be much more descriptive if something breaks. 

Moreover, instead of putting the expectations in a comment (i.e., CONF + QUERY) just create manually the expected List and base your assertion there.